ruby/lib/ci/queue/redis/build_record.rb

```diff
@@ -0,0 +1,63 @@
+module CI
+  module Queue
+    module Redis
+      class BuildRecord
+        def initialize(queue, redis, config)
+          @queue = queue
+          @redis = redis
+          @config = config
+        end
+
+        def progress
+          @queue.progress
+        end
+
+        def queue_exhausted?
+          @queue.exhausted?
+        end
```

Comment:
There is some duplication in the methods above with `CI::Queue::BuildRecord`, could we inherit from it?

Reply:
Yeah we could probably share a base class or a module. I'll keep that in mind of a future refactor.